lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -117,21 +120,21 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
             <div>
               <div class="input-label">Storage</div>
               <div class="my-2 space-y-1 text-sm">
-                <%= label  class: "flex items-center gap-2 text-gray-600" do %>
+                <%= label class: "flex items-center gap-2 text-gray-600" do %>
                   <%= radio_button(f, :store, "session", checked: @data["store"] == "session") %> only this session
                 <% end %>
+
                 <%= label class: "flex items-center gap-2 text-gray-600" do %>
                   <%= radio_button(f, :store, "app", checked: @data["store"] == "app") %> in the Livebook app
                 <% end %>
+
                 <%= if Livebook.Config.feature_flag_enabled?(:hub) do %>
-                  <%= label class: "flex items-center gap-2 text-gray-600" do %>
-                    <%= radio_button(f, :store, "hub",
-                      disabled: @hubs == [],
-                      checked: @data["store"] == "hub"
-                    ) %> in the Hub
-                  <% end %>
-                  <%= if @data["store"] == "hub" do %>
-                    <%= select(f, :hub_id, hubs_options(@hubs, @data["hub_id"]), class: "input") %>
+                  <%= for hub <- @hubs do %>
+                    <%= label class: "flex items-center gap-2 text-gray-600" do %>
+                      <%= radio_button(f, :store, "hub:#{hub.id}",
+                        checked: @data["store"] == "hub:#{hub.id}"
```

Comment:
as mentioned above I think ideally we'd have a changeset so this would become:

```elixir
radio_button(f, :origin, "hub-#{hub.id}")
                ^^^
```

except... it wouldn't work. The value that would be stored under the hood would be from the custom Ecto type, `{:hub, id}`, and it would crash displaying the button because there's no Phoenix.HTML.Safe for tuple. So we would have to manually build the input, `<input id={input_id(...)} checked={...} ...`. Alternatively we could do either of these things:

1. Change `{:hub, id}` to `"hub-id"`
2. Have a `origin_string` (or indeed, :store, as it is called today) virtual field on the changeset that works on strings
3. Change SecretOrigin `{:hub, id}` to `%SecretOrigin.HubID{id: id}` and implement html safe for it.
4. Change Phoenix.HTML.Form input_value to use Ecto.Type.dump for custom type. Not sure it is general enough.

but tbh neighter of these sounds fantastic so having an `<input>` is probably the best way to go anyway.

Reply:
LGTM using the input strategy